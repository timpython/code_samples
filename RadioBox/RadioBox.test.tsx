import React from 'react';
import { shallow } from 'enzyme';
import { RadioBox } from '.';

const options = [
  {
    label: 'Cookie',
    value: 0,
    content: <div>Cookie Description</div>,
  },
  {
    label: 'LocalStorage',
    value: 1,
    content: <div>Local Storage Description</div>,
  },
];

const selection = 0;

describe('RadioBox', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <RadioBox options={options} selection={selection} name="someName" containerClassName="someContainerClassName" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('have correct listener', () => {
    const onChange = () => {};
    const wrapper = shallow(<RadioBox options={options} selection={selection} onChange={onChange} name="someName" />);

    const callback = wrapper
      .find('RadioBoxOption')
      .first()
      .props().onChange;

    expect(onChange).toEqual(callback);
    expect(wrapper).toMatchSnapshot();
  });
});
