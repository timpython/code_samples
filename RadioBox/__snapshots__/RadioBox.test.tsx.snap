// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioBox have correct listener 1`] = `
<div>
  <div
    className="body"
  >
    <RadioBoxOption
      checked={true}
      content={
        <div>
          Cookie Description
        </div>
      }
      key="Cookie"
      label="Cookie"
      name="someName"
      onChange={[Function]}
      value={0}
    />
    <RadioBoxOption
      checked={false}
      content={
        <div>
          Local Storage Description
        </div>
      }
      key="LocalStorage"
      label="LocalStorage"
      name="someName"
      onChange={[Function]}
      value={1}
    />
  </div>
</div>
`;

exports[`RadioBox renders correctly 1`] = `
<div
  className="someContainerClassName"
>
  <div
    className="body"
  >
    <RadioBoxOption
      checked={true}
      content={
        <div>
          Cookie Description
        </div>
      }
      key="Cookie"
      label="Cookie"
      name="someName"
      onChange={[Function]}
      value={0}
    />
    <RadioBoxOption
      checked={false}
      content={
        <div>
          Local Storage Description
        </div>
      }
      key="LocalStorage"
      label="LocalStorage"
      name="someName"
      onChange={[Function]}
      value={1}
    />
  </div>
</div>
`;
